crates/polars-core/src/series/any_value.rs

```diff
@@ -235,6 +235,10 @@ impl Series {
                 any_values_to_decimal(av, *precision, *scale)?.into_series()
             },
             DataType::List(inner) => any_values_to_list(av, inner, strict)?.into_series(),
+            #[cfg(feature = "dtype-array")]
+            DataType::Array(inner, size) => any_values_to_list(av, inner, strict)?
```

Comment:
This will allocate list offsets where we don't have to.

We should add a `any_values_to_array`. This one can piggy back on any_values to series creation and then store the array in the an arrow fixedsize list.